File: java - oop/src/main/java/BaseAdvertising.java
public class BaseAdvertising {
    protected int id;
    protected int views_count;
    protected int clicks_count;

    protected int getClicks() {
        return clicks_count;
    }

    protected int getViews() {
        return views_count;
    }

    protected void incViews() {
        views_count++;
    }

    protected void incClicks() {
        clicks_count++;
    }

    public String describeMe() {
        return "'This is 'Base Advertising class'. Other classes use my fields(views_count & clicks_count) and methods. Classes use functions for getting and incrementing fields in this class. This class make this code cleaner!')";
    }
}

File: python - oop/pythonProject1/BaseAdvertising.py
class BaseAdvertising:
    def __init__(self):
        self.views_count = 0
        self.clicks_count = 0

    @staticmethod
    def describeMe() -> str:
        print(
            'This is \'Base Advertising\'. Other classes use my fields(views_count & clicks_count) and methods. Classes use functions for getting and increamenting fields in this class. This class make this code cleaner!')

    def getClicks(self) -> int:
        return self.clicks_count

    def getViews(self) -> int:
        return self.views_count

    def incViews(self):
        self.views_count += 1

    def incClicks(self):
        self.clicks_count += 1
